src/core/server/http/router/response_adapter.ts

```diff
@@ -72,6 +72,9 @@ export class HapiResponseAdapter {
   }
 
   private toHapiResponse(kibanaResponse: KibanaResponse) {
+    if (this.isAlreadyAborted()) {
+      return this.responseToolkit.close;
+    }
```

Comment:
@mshustov If I understand correctly, this is fixing a significant bug, isn't it? Should we also add an integration test to cover the fix?

Reply:
This change doesn't fix the problem, unfortunately. A fix should be done on the Hapi level. See https://github.com/hapijs/hapi/issues/4294 for a detailed description.